File: romanToInt.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
    const map = new Map()
    
    map.set('I', 1)
    map.set('V', 5)
    map.set('X', 10)
    map.set('L', 50)
    map.set('C', 100)
    map.set('D', 500)
    map.set('M', 1000)

    let sum = 0

    for (let i = 0; i < s.length; i++) {
        if (map.get(s[i]) < map.get(s[i + 1])) {
            sum = sum - map.get(s[i])
        } else {
            sum = sum + map.get(s[i])
        }
    }
    return sum
};

console.log(romanToInt('IV'))

File: romanToInt.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution13 {
    public static int romanToInt(String s) {
        Map<Character, Integer> map = new HashMap<>();

        List<Character> list = new ArrayList<>();

        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        int sum = 0;
        for (int i = 0; i < s.length()-1; i++) {
            if (map.get(s.charAt(i))<map.get(s.charAt(i+1))){
                sum-=map.get(s.charAt(i));
            }
            else {
                sum+=map.get(s.charAt(i));
            }
        }

        return sum+map.get(s.charAt(s.length()-1));
    }

    public static void main(String[] args) {
        System.out.println(romanToInt("III"));
    }
}
